portable/NetworkInterface/ThirdParty/MSP432/NetworkInterface.c

```diff
@@ -0,0 +1,987 @@
+                         uint32_t ul32BufLen )
+{
+    uint8_t bufferTX[ ETH_TX_BUFFER_SIZE ];
+
+    ulg_ui32TxDescIndex++;
+
+    if( ulg_ui32TxDescIndex == NUM_TX_DESCRIPTORS )
+    {
+        ulg_ui32TxDescIndex = 0;
+    }
+
+    /* The copy needs to be done here since directly assigning a pointer does not seem to work
+     * and causes the transmitter to stop.  Some testing indicates that that this can be quicker than
+     *  simply assigning the pointer for smaller packets. */
+    memcpy( bufferTX, pui8Buf, ul32BufLen );
+
+    g_psTxDescriptor[ ulg_ui32TxDescIndex ].ui32Count = ul32BufLen;
+    g_psTxDescriptor[ ulg_ui32TxDescIndex ].pvBuffer1 = bufferTX;
+    g_psTxDescriptor[ ulg_ui32TxDescIndex ].ui32CtrlStatus =
+        ( DES0_TX_CTRL_LAST_SEG | DES0_TX_CTRL_FIRST_SEG |
+          DES0_TX_CTRL_INTERRUPT | DES0_TX_CTRL_IP_ALL_CKHSUMS |
+          DES0_TX_CTRL_CHAINED | DES0_TX_CTRL_OWN );
+
+    EMACTxDMAPollDemand( EMAC0_BASE );
+    return( ul32BufLen );
+}
+
+
+/* Function to process the received packet */
+uint32_t processReceivedPacket( void )
+{
+    uint32_t ul32FrameLen;
+
+    ul32FrameLen = 0;
+
+    if( !( g_psRxDescriptor[ ulg_ui32RxDescIndex ].ui32CtrlStatus & DES0_RX_CTRL_OWN ) )
+    {
+        /* Does it have a valid frame? */
+        if( !( g_psRxDescriptor[ ulg_ui32RxDescIndex ].ui32CtrlStatus &
+               DES0_RX_STAT_ERR ) )
+        {
+            if( g_psRxDescriptor[ ulg_ui32RxDescIndex ].ui32CtrlStatus &
+                DES0_RX_STAT_LAST_DESC )
+            {
+                /* get the frame length */
+                ul32FrameLen =
+                    ( ( g_psRxDescriptor[ ulg_ui32RxDescIndex ].ui32CtrlStatus &
+                        DES0_RX_STAT_FRAME_LENGTH_M ) >>
+                      DES0_RX_STAT_FRAME_LENGTH_S );
+
+                /* call the function that sends the data to the task */
+                applicationProcessFrameRX( ul32FrameLen,
+                                           g_psRxDescriptor[ ulg_ui32RxDescIndex ].pvBuffer1, ulg_ui32RxDescIndex );
+            }
+        }
+
+        /* Move on to the next RX packet */
+        ulg_ui32RxDescIndex++;
+
+        if( ulg_ui32RxDescIndex == NUM_RX_DESCRIPTORS )
+        {
+            ulg_ui32RxDescIndex = 0;
+        }
+    } /* end if */
+
+    /* return the frame length */
+    return ul32FrameLen;
+}
+
+
+/* This function passes the framelength and the RX buffer into the FreeRTOS task.
+ * The function is called from the ISR and therefore must use portYIELD_FROM_ISR() */
+void applicationProcessFrameRX( uint32_t ul32FrameLen,
+                                uint8_t * uc8Buf,
+                                uint32_t ulindex )
+{
+    BaseType_t rv;
+    struct NetworkInterfaceDataIn NIDataIn;
+
+    NIDataIn.ucbuff = uc8Buf;
+    NIDataIn.ulbuff_siz = ul32FrameLen;
+    NIDataIn.ulindx = ulindex;
+
+    /* send the data into the queue */
+    rv = xQueueSendFromISR( xQueueRX, &NIDataIn, &xHigherPriorityTaskWokenRX );
```

Comment:
Oops, `xHigherPriorityTaskWokenRX` is cleared when the application starts. You driver will never clear it again.
If is declared as a global variable, is it cleared elsewhere?
~~~c
     /* send the data into the queue */
     rv = xQueueSendFromISR( xQueueRX, &NIDataIn, &xHigherPriorityTaskWokenRX );

     if( rv == pdTRUE )
     {
         vTaskNotifyGiveFromISR( xTaskToNotifyEthernetRX, &xHigherPriorityTaskWokenRX );
-        portYIELD_FROM_ISR( xHigherPriorityTaskWokenRX );
     }
     else
     {
         DMAFreeDescriptorRX( ulindex );
     }
+    portYIELD_FROM_ISR( xHigherPriorityTaskWokenRX );
~~~
Suppose that `xQueueSendFromISR()` sets `xHigherPriorityTaskWokenRX ` while returning `pdFALSE`, then `portYIELD_FROM_ISR()` will not be called?